From e66561316f6f0b825e647fd2d14c6d00e8905bd1 Mon Sep 17 00:00:00 2001
From: Xinhuang Li <buluess.li@rock-chips.com>
Date: Fri, 5 Jan 2018 11:03:43 +0800
Subject: [PATCH] video: rockchip: vpu: clk_core is not necessary

if the combo is avsd and vdpu,clk_core is not necessary;
others case,the DTS will set the clk_core and here will
get the clk as well.

Change-Id: I5e33375e4655763791b9c4418bce7cbe0f5ac530
Signed-off-by: Xinhuang Li <buluess.li@rock-chips.com>
---
 drivers/video/rockchip/vcodec/vcodec_service.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/vcodec/vcodec_service.c b/drivers/video/rockchip/vcodec/vcodec_service.c
index e2ac3c58ced5..6aa7b317b47c 100644
--- a/drivers/video/rockchip/vcodec/vcodec_service.c
+++ b/drivers/video/rockchip/vcodec/vcodec_service.c
@@ -622,7 +622,7 @@ static int vpu_get_clk(struct vpu_service_info *pservice)
 		pservice->clk_core = devm_clk_get(dev, "clk_core");
 		if (IS_ERR(pservice->clk_core)) {
 			dev_err(dev, "failed on clk_get clk_core\n");
-			return -1;
+			pservice->clk_core = NULL;
 		} else {
 			pservice->clk_core_default_rate =
 				clk_get_rate(pservice->clk_core);
-- 
2.35.3

